src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Approve.tsx

```diff
@@ -143,6 +144,7 @@ export const Approve: React.FC<ApproveProps> = ({ accountId, onNext }) => {
       const contract = getOrCreateContractByType({
         address: fromAssetId(assetId).assetReference,
         type: ContractType.ERC20,
+        chainId: KnownChainIds.EthereumMainnet,
```

Comment:
Can also be Avalanche and BSC, this needs to be programmatic:
<img width="388" alt="image" src="https://github.com/shapeshift/web/assets/17035424/3b848c75-4666-47e1-93b3-8ac05f82b261">